guacamole/src/main/webapp/app/navigation/services/userPageService.js

```diff
@@ -57,12 +57,24 @@ angular.module('navigation').factory('userPageService', ['$injector',
      *     A map of all root connection groups visible to the current user,
      *     where each key is the identifier of the corresponding data source.
      *
+     * @param {Object.<String, PermissionSet>} permissions
+     *     A map of all permissions granted to the current user, where each
+     *     key is the identifier of the corresponding data source.
+     *
+     * @param {Object.<String,
+     *
      * @returns {PageDefinition}
      *     The user's home page.
      */
-    var generateHomePage = function generateHomePage(rootGroups) {
+    var generateHomePage = function generateHomePage(rootGroups, permissions) {
 
         var homePage = null;
+        var settingsPages = generateSettingsPages(permissions);
+
+        // If we have more than one setting page, return the main home page
```

Comment:
Better, but the critical fact we want to highlight here is that _more than one settings page indicates that the user has administrative privileges_, and thus redirecting to a single connection would hurt the user experience. The fact that this comparison checks for more than one page is clear from the `> 1`, but the meaning behind that check is non-obvious.

Reply:
Okay, added some more verbiage there.